### PR TITLE
JDK-8280274: Guard printing code of Compile::print_method in PRODUCT

### PR DESCRIPTION
**Pull request by tobiasholenstein**
In `Compile::print_method(CompilerPhaseType cpt, Node* n, int level)` the code 
``` 
ResourceMark rm; 
stringStream ss; 
ss.print_raw(CompilerPhaseTypeHelper::to_string(cpt)); 
if (n != NULL) { 
  ss.print(": %d %s ", n->_idx, NodeClassNames[n->Opcode()]); 
} else { 
  ss.print_raw(": NULL"); 
} 
``` 
should to be guarded by a `#ifndef PRODUCT` since the usage of the string is also guarded in `Compile::print_method_impl`

Tested on Tier1-3

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed

### Issue
 * [JDK-8280274](https://bugs.openjdk.java.net/browse/JDK-8280274): Guard printing code of Compile::print_method in PRODUCT

### Reviewers
 * [Tobias Hartmann](https://openjdk.java.net/census#thartmann) (@TobiHartmann - **Reviewer**)
 * [Nils Eliasson](https://openjdk.java.net/census#neliasso) (@neliasso - **Reviewer**) ⚠️ Review applies to dce24e2aae4fe355952b34e5bc514891bcc9a986

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.java.net/jdk pull/7158/head:pull/7158` \
`$ git checkout pull/7158`

Update a local copy of the PR: \
`$ git checkout pull/7158` \
`$ git pull https://git.openjdk.java.net/jdk pull/7158/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 7158`

View PR using the GUI difftool: \
`$ git pr show -t 7158`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.java.net/jdk/pull/7158.diff">https://git.openjdk.java.net/jdk/pull/7158.diff</a>

</details>

### CODE REVIEW

File: src/hotspot/share/opto/compile.hpp [hunk 1]
```diff
@@ -656,8 +656,8 @@ class Compile : public Phase {
   bool should_print_igv(int level);
 
   void print_method(CompilerPhaseType cpt, int level);
-  void print_method(CompilerPhaseType cpt, Node* n, int level = 3);
-  void print_method_impl(CompilerPhaseType cpt, const char *name, int level);
+  void print_method(CompilerPhaseType cpt, Node* n, int level);
+  void print_method_impl(CompilerPhaseType cpt, NOT_PRODUCT(const char *name COMMA) int level);
```
Comment (TobiHartmann) [inline src/hotspot/share/opto/compile.hpp:660]:
```suggestion
  void print_method_impl(CompilerPhaseType cpt, NOT_PRODUCT(const char* name COMMA) int level);
```

File: src/hotspot/share/opto/compile.cpp [hunk 2]
```diff
@@ -4827,10 +4828,11 @@ void Compile::print_method(CompilerPhaseType cpt, Node* n, int level) {
   } else {
     ss.print_raw(": NULL");
   }
-  C->print_method_impl(cpt, ss.as_string(), level);
+#endif
+  C->print_method_impl(cpt, NOT_PRODUCT(ss.as_string() COMMA) level);
 }
 
-void Compile::print_method_impl(CompilerPhaseType cpt, const char *name, int level) {
+void Compile::print_method_impl(CompilerPhaseType cpt, NOT_PRODUCT(const char *name COMMA) int level) {
```
Comment (TobiHartmann) [inline src/hotspot/share/opto/compile.cpp:4835]:
```suggestion
void Compile::print_method_impl(CompilerPhaseType cpt, NOT_PRODUCT(const char* name COMMA) int level) {
```

**Review by TobiHartmann — COMMENTED** [review]
Would it also be an option to move that code into `print_method_impl`?

**Review by neliasso — APPROVED** [review]
Looks good.

**Comment by tobiasholenstein** [comment]
> Would it also be an option to move that code into `print_method_impl`?

Unfortunately, that would also change the printing of `Compile::print_method(CompilerPhaseType cpt, int level)` where `print_method_impl` is also used.

**Review by TobiHartmann — APPROVED** [review]
Right, makes sense. Looks good to me.

**Comment by tobiasholenstein** [comment]
Thanks for the reviews, @TobiHartmann and @neliasso 

/integrate

**Pull request closed without merging**